model.js

```diff
@@ -33,4 +33,23 @@ async function selectArticleByID(articleID) {
   }
 }
 
-module.exports = { selectAllTopics, selectEndpoints, selectArticleByID };
+async function selectAllArticles() {
+  try {
+    const query = await db.query(
+      `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url,
+        (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id)::integer AS comment_count
+      FROM articles
+      ORDER BY created_at desc`
+    );
+    return query.rows;
```

Comment:
If this works then I'm happy with it. I would have used a count, group by, and left join instead though. Please let me know if you'd like me to expand upon this.